test/unit-tests/components/views/settings/encryption/ChangeRecoveryKey-test.tsx

```diff
@@ -120,27 +122,66 @@ describe("<ChangeRecoveryKey />", () => {
             mocked(matrixClient.getCrypto()!).bootstrapSecretStorage.mockRejectedValue(new Error("can't bootstrap"));
 
             const user = userEvent.setup();
-            renderComponent(false);
+            const { getByRole, getByText, getByTitle } = renderComponent(false);
 
             // Display the recovery key to save
-            await waitFor(() => user.click(screen.getByRole("button", { name: "Continue" })));
+            await waitFor(() => user.click(getByRole("button", { name: "Continue" })));
             // Display the form to confirm the recovery key
-            await waitFor(() => user.click(screen.getByRole("button", { name: "Continue" })));
+            await waitFor(() => user.click(getByRole("button", { name: "Continue" })));
 
-            await waitFor(() => expect(screen.getByText("Enter your recovery key to confirm")).toBeInTheDocument());
+            await waitFor(() => expect(getByText("Enter your recovery key to confirm")).toBeInTheDocument());
 
-            const finishButton = screen.getByRole("button", { name: "Finish set up" });
-            const input = screen.getByTitle("Enter recovery key");
+            const finishButton = getByRole("button", { name: "Finish set up" });
+            const input = getByTitle("Enter recovery key");
             await userEvent.type(input, "encoded private key");
+
+            await waitFor(() => {
+               expect(finishButton).not.toBeDisabled();
+            });
+
+            jest.spyOn(Modal, "createDialog");
+
             await user.click(finishButton);
 
-            await screen.findByText("Failed to set up secret storage");
-            await screen.findByText("Error: can't bootstrap");
+            expect(Modal.createDialog).toHaveBeenCalledWith(ErrorDialog, {
```

Comment:
@andybalaam Noting this change. This test was causing an ErrorDialog to popup and that was apparently persisting to the next test. Other tests seem to mock out createDialog for this sort of error testing, so I suppose it's been encountered before.

Anyway, this seemed to make it all work.